modules/dns_nsupdate/dns_nsupdate_main.rb

```diff
@@ -55,23 +55,23 @@ def do_create(id, value, type)
     end
 
     def remove_a_record(fqdn)
+      get_address!(fqdn)
       do_remove(fqdn, "A")
     end
 
     def remove_aaaa_record(fqdn)
+      get_address!(fqdn)
       do_remove(fqdn, "AAAA")
     end
 
     def remove_ptr_record(ip)
+      get_name!(ip)
       do_remove(ip, "PTR")
     end
 
     def do_remove(id, type)
       nsupdate_connect
-
-      raise Proxy::Dns::NotFound.new("Cannot find DNS entry for #{id}") unless dns_find(id)
       nsupdate "update delete #{id} #{type}"
-
       nsupdate_disconnect
       nil
     end
```

Comment:
Part two of the ticket was about a missing `ensure` block here, so the omshell processes were leaking. Either add it here or separate into a new ticket if you prefer.